terraform/modules/ecs_module/ecs.tf

```diff
@@ -0,0 +1,125 @@
+provider "aws" {
+  region = var.aws_region
+}
+
+resource "aws_ecs_cluster" "aws-ecs" {
+  name = var.app_name
+}
+
+data "aws_ami" "ecs-ami" {
+  most_recent = true
+  filter {
+    name   = "name"
+    values = ["amzn2-ami-ecs-hvm-2.0.*"]
+  }
+  filter {
+    name   = "architecture"
+    values = ["x86_64"]
+  }
+  owners = ["amazon"]
+}
+
+variable "aws_ecs_ami_override" {
+  default     = ""
+  description = "Machine image to use for ec2 instances"
+}
+
+locals {
+  aws_ecs_ami = var.aws_ecs_ami_override == "" ? data.aws_ami.ecs-ami.id : var.aws_ecs_ami_override
+}
+
+resource "aws_iam_role" "ecs-cluster-runner-role" {
+  name               = "${var.app_name}-cluster-runner-role"
+  assume_role_policy = data.aws_iam_policy_document.instance-assume-role.json
+}
+
+data "aws_caller_identity" "current" {}
+
+data "aws_iam_policy_document" "ecs-cluster-runner-policy" {
+  statement {
+    actions   = ["ec2:Describe*", "ecr:Describe*", "ecr:BatchGet*"]
+    resources = ["*"]
+  }
+  statement {
+    actions   = ["ecs:*"]
+    resources = ["arn:aws:ecs:${var.aws_region}:${data.aws_caller_identity.current.account_id}:service/${var.app_name}/*"]
+  }
+}
+
+resource "aws_iam_role_policy" "ecs-cluster-runner-role-policy" {
+  name   = "${var.app_name}-cluster-runner-policy"
+  role   = aws_iam_role.ecs-cluster-runner-role.name
+  policy = data.aws_iam_policy_document.ecs-cluster-runner-policy.json
+}
+
+resource "aws_iam_instance_profile" "ecs-cluster-runner-profile" {
+  name = "${var.app_name}-cluster-runner-iam-profile"
+  role = aws_iam_role.ecs-cluster-runner-role.name
+}
+
+data "template_file" "user_data_cluster" {
+  template = file("${path.module}/templates/cluster_user_data.sh")
+  vars = {
+    ecs_cluster = aws_ecs_cluster.aws-ecs.name
+  }
+}
+
+resource "aws_instance" "ecs-cluster-runner" {
```

Comment:
🚫 **[tfsec]** <[AVD-AWS-0028](https://aquasecurity.github.io/tfsec/v1.27.6/checks/aws/ec2/enforce-http-token-imds/)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Instance does not require IMDS access to require a token